### PR TITLE
--url-file: skip long and empty lines instead of successfully aborting

### PR DESCRIPTION
**Pull request by emanuele6**
This fixes a regression from aa62bb1 that caused trurl to abort successfully at the first empty (but not blank) line instead of simply skipping it.

This patch also makes trurl skip too long lines instead of successfully aborting when encountering one.

It also avoids calling fgets() after EOF, and reports read errors caused by getc()/fgets() if any.

I added a testfiles/ directory in which to add --url-file files to use in tests, and some tests that check that empty lines are ignored, long lines are reported and skipped, and spaces and tabs, and final carriage return are trimmed off the end lines.

### CODE REVIEW

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
tests.json conflicts now though, do you mind fixing that?

**Comment by emanuele6** [comment]
Done

File: tests.json [hunk 1]
```diff
@@ -1729,6 +1729,32 @@
             "stdout": "emanuele6://curl.se/trurl\nhttps://example.org/\n",
             "returncode": 0,
             "stderr": "trurl note: No host part in the URL []\n"
+    },
```
Comment (bagder) [inline tests.json:1732]: There's a missing close brace here isn't it?

**Review by bagder — APPROVED** [review]

**Comment by emanuele6** [comment]
The commit had the wrong name for some reason :/
